src/test/java/com/ironcorelabs/tenantsecurity/kms/v1/DeterministicCryptoUtilsTest.java

```diff
@@ -0,0 +1,290 @@
+  }
+
+  public void encryptFieldBatchRoundtrip() throws Exception {
+    Map<String, DeterministicPlaintextField> batch = getBatchMap(5);
+    BatchResult<DeterministicEncryptedField> encrypted = DeterministicCryptoUtils
+        .encryptFieldBatch(batch, getDeriveKeyResponse(), Executors.newWorkStealingPool());
+    assertEquals(encrypted.getSuccesses().size(), 5);
+    assertEquals(encrypted.getFailures().size(), 0);
+    BatchResult<DeterministicPlaintextField> decrypted = DeterministicCryptoUtils.decryptFieldBatch(
+        encrypted.getSuccesses(), getDeriveKeyResponse(), Executors.newWorkStealingPool());
+    assertEquals(decrypted.getSuccesses().size(), 5);
+    assertEquals(decrypted.getFailures().size(), 0);
+    assertEqualBytes(decrypted.getSuccesses().get("0").getPlaintextField(),
+        batch.get("0").getPlaintextField());
+  }
+
+  public void encryptFieldBatchExceptionsTest() throws Exception {
+    Map<String, DeterministicPlaintextField> batch = new HashMap<>();
+    batch.put("0", new DeterministicPlaintextField("aaaaaa".getBytes(), "deriv", "secret"));
+    batch.put("1", new DeterministicPlaintextField("aaaaaa".getBytes(), "badPath", "secret"));
+
+    BatchResult<DeterministicEncryptedField> encrypted = DeterministicCryptoUtils
+        .encryptFieldBatch(batch, getDeriveKeyResponse(), Executors.newWorkStealingPool());
+    assertEquals(encrypted.getSuccesses().size(), 1);
+    assertEquals(encrypted.getFailures().size(), 1);
+    TenantSecurityException ex = encrypted.getFailures().get("1");
+    assertTrue(ex instanceof TenantSecurityException);
+    assertEquals(ex.getErrorCode(), TenantSecurityErrorCodes.UNKNOWN_ERROR);
+    assertEquals(ex.getMessage(), "TSP failed to derive keys.");
+  }
+
+  public void generateSearchTermsBatchTest() throws Exception {
+    Map<String, DeterministicPlaintextField> batch = getBatchMap(5);
+    BatchResult<DeterministicEncryptedField[]> encrypted = DeterministicCryptoUtils
+        .generateSearchTermsBatch(batch, getDeriveKeyResponse(), Executors.newWorkStealingPool());
+    assertEquals(encrypted.getSuccesses().size(), 5);
+    assertEquals(encrypted.getFailures().size(), 0);
+    assertEquals(encrypted.getSuccesses().get("0").length, 2);
+    assertEquals(encrypted.getSuccesses().get("1").length, 2);
+  }
+
+  @Test(expectedExceptions = java.util.concurrent.ExecutionException.class,
+      expectedExceptionsMessageRegExp = ".*No current.*")
+  public void encryptFailsWithNoCurrent() throws Exception {
+    DerivedKey[] noCurrentDerivedKeys = {new DerivedKey(
+        "AUvYQZVvGpqalGZyO7Sy5WSsJ9KqOkwP/jlQnvORy/hZVU1pTCLefEPKJ4mShUfdKOKbECMpuf7YpR9+CNwuEQ==",
+        6, false)};
+    DeterministicCryptoUtils.encryptField(plaintextField, noCurrentDerivedKeys).get();
+  }
+
+  @Test(expectedExceptions = java.util.concurrent.ExecutionException.class,
+      expectedExceptionsMessageRegExp = ".*not valid base64.*")
+  public void encryptFailsWithInvalidBase64() throws Exception {
```

Comment:
This test doesn't check if the exception is thrown by calling the function or if it's called by doing the `.get`. It's an important distinction we should check.

Reply:
Added another test without the `.get`